src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -356,67 +364,35 @@ else if ( new File(arg).exists() ) {
         Utils.nonNull(fileName, "file name is null");
 
         final Path inputPath = IOUtils.getPath(fileName);
         final List<GenomeLoc> ret = new ArrayList<>();
 
-        /**
-         * First try to read the file as a Picard interval file since that's well structured --
-         * we'll fail quickly if it's not a valid file.
-         */
-        boolean isPicardInterval = false;
-        try {
-            // Note: Picard will skip over intervals with contigs not in the sequence dictionary
-            final IntervalList il = IntervalList.fromPath(inputPath);
-            isPicardInterval = true;
-
-            for (final Interval interval : il.getIntervals()) {
-                // The current Agilent exome interval list is off-by-one on all end positions. Until this is fixed we
-                // need to tolerate intervals where the end is one before the start. We should remove this once a
-                // corrected version of the interval list is released. This is tracked in:
-                // https://github.com/broadinstitute/gatk/issues/2089
-                if (interval.getStart() - interval.getEnd() == 1 ) {
-                    logger.warn("Ignoring possibly incorrectly converted length 1 interval : " + interval);
-                }
-                else if ( glParser.isValidGenomeLoc(interval.getContig(), interval.getStart(), interval.getEnd(), true)) {
-                    ret.add(glParser.createGenomeLoc(interval.getContig(), interval.getStart(), interval.getEnd(), true));
-                } else {
-                    throw new UserException(inputPath.toUri() +  " has an invalid interval : " + interval) ;
+        try (final PathLineIterator reader = new PathLineIterator(inputPath)) {
+            for (final String line : reader) {
+                final String trimmedLine = line.trim();
+                if (!trimmedLine.isEmpty()) {
+                    ret.add(glParser.parseGenomeLoc(trimmedLine));
                 }
             }
-        }
-        // if that didn't work, try parsing file as a GATK interval file
-        catch (final Exception e) {
-            if ( isPicardInterval ) // definitely a picard file, but we failed to parse
-            {
-                throw new UserException.CouldNotReadInputFile(inputPath, e);
-            } else {
-                try (PathLineIterator reader = new PathLineIterator(inputPath)) {
-                    for (final String line : reader) {
-                        final String trimmedLine = line.trim();
-                        if (!trimmedLine.isEmpty()) {
-                            ret.add(glParser.parseGenomeLoc(trimmedLine));
-                        }
-                    }
-                }
+            if (ret.isEmpty()) {
+                throw new UserException.MalformedFile(inputPath, "It contains no intervals.");
             }
+            return ret;
+        } catch (final UserException e){
+            throw e;
```

Comment:
This block seems unnecessary ? It might be better to just rely on the next catch block, which will always include the inputPath name.

Reply:
I think I thought it might replace a more specific error with a less specific one.  Since userexceptions don't show the stack trace you'd lose the underlying one in most cases.